src/rgw/rgw_common.h

```diff
@@ -2079,6 +2079,29 @@ static inline void buf_to_hex(const unsigned char* const buf,
   }
 }
 
+/* A variant of buf_to_hex dedicated to handling ceph::bufferlist which
+ * size can't be deduced at compile-time. This translates into necessity
+ * to use heap as VLA: 1) is a GCC's extension and 2) can be dangerous.
+ * The procedure is optimized to avoid extra linearisation of incoming
+ * buffers. */
+static inline std::string buf_to_hex(const ceph::bufferlist& bl)
+{
+  std::string hexed;
+  hexed.resize(bl.length() * 2);
+
+  size_t consumed = 0;
+  for (const auto& buf : bl.buffers()) {
+    /* NOTE(rzarzynski): yes, we're writing directly to underlying buffer
```

Comment:
Please remember to remove name `rzarzynski`.
May be later on PR would be raised for removal.